main.tf

```diff
@@ -1,13 +1,78 @@
-resource "random_integer" "example" {
+resource "kubernetes_service_account" "service_account" {
   count = module.this.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  metadata {
+    name        = coalesce(var.deployment_name, module.this.id, "tfc-agent")
+    namespace   = coalesce(var.kubernetes_namespace, var.namespace, "default")
+    annotations = var.service_account_annotations
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "kubernetes_deployment" "tfc_cloud_agent" {
+  count = module.this.enabled ? 1 : 0
+
+  metadata {
+    name      = coalesce(var.deployment_name, module.this.id, "tfc-agent")
```

Comment:
Should we add an option to create the namespace?

Reply:
It looks a bit tricky in Terraform (since you have to check whether it already exists), but I'll give it a shot 👍

Reply:
Oh no, that's the job of the user. If the user specifies `namespace_creation_enabled` to `true` and the namespace exists, then it should error. 

On the other hand, if they specify `false`, and the namespace does not exist, the deployment creation should fail.